compiler/wasm/src/compile.rs

```diff
@@ -24,14 +28,26 @@ extern "C" {
     pub type StringArray;
 }
 
+#[derive(Deserialize)]
+struct DependencyGraph {
+    root_dependencies: Vec<String>,
+    library_dependencies: HashMap<String, Vec<String>>,
+}
+
 #[wasm_bindgen]
 pub fn compile(
     entry_point: String,
     contracts: Option<bool>,
-    dependencies: Option<StringArray>,
+    dependency_graph: JsValue,
 ) -> Result<JsValue, JsCompileError> {
     console_error_panic_hook::set_once();
 
+    let dependency_graph: DependencyGraph =
+        <JsValue as JsValueSerdeExt>::into_serde(&dependency_graph).unwrap_or(DependencyGraph {
+            root_dependencies: vec![],
+            library_dependencies: HashMap::new(),
+        });
+
```

Comment:
This will result in us ignoring any malformed values for `dependency_graph` which will cause misleading error messages when the compiler can't find these dependencies. We should throw a proper error here if we have a bad dependency graph (or preferably we have type safety so that a user can't pass an unreadable dependency graph.